sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpLogOptions.java

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.implementation.http.policy.HttpRequestLogger;
+import com.generic.core.implementation.http.policy.HttpResponseLogger;
+import com.generic.core.util.configuration.Configuration;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.generic.core.util.configuration.Configuration.getGlobalConfiguration;
+
+/**
+ * The log configurations for HTTP messages.
+ */
+public final class HttpLogOptions {
+    private HttpLogDetailLevel logLevel;
+    private Set<String> allowedHeaderNames;
+    private Set<String> allowedQueryParamNames;
+    private boolean prettyPrintBody;
+    private HttpRequestLogger requestLogger;
+    private HttpResponseLogger responseLogger;
+    private static final List<String> DEFAULT_HEADERS_ALLOWLIST = Arrays.asList(
+        "traceparent",
+        "Accept",
+        "Cache-Control",
+        "Connection",
+        "Content-Length",
+        "Content-Type",
+        "Date",
+        "ETag",
+        "Expires",
+        "If-Match",
+        "If-Modified-Since",
+        "If-None-Match",
+        "If-Unmodified-Since",
+        "Last-Modified",
+        "Pragma",
+        "Request-Id",
+        "Retry-After",
+        "Server",
+        "Transfer-Encoding",
+        "User-Agent",
+        "WWW-Authenticate"
+    );
+
+    private static final List<String> DEFAULT_QUERY_PARAMS_ALLOWLIST = Collections.singletonList(
+        "api-version"
+    );
+
+    /**
+     * Creates a new instance that does not log any information about HTTP requests or responses.
+     */
+    public HttpLogOptions() {
+        logLevel = HttpLogDetailLevel.ENVIRONMENT_HTTP_LOG_DETAIL_LEVEL;
+        allowedHeaderNames = new HashSet<>(DEFAULT_HEADERS_ALLOWLIST);
+        allowedQueryParamNames = new HashSet<>(DEFAULT_QUERY_PARAMS_ALLOWLIST);
```

Comment:
But how will then the user be able to see the default allowed headers list before doing the `add` .
It would be empty in that case since we deferred  creating the HashSet until add* methods are called.